airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/test/steps/common.py

```diff
@@ -668,3 +660,103 @@ async def _build_regression_test_container(self, target_container_id: str) -> Co
 
         container = container.with_exec(["poetry", "lock", "--no-update"]).with_exec(["poetry", "install"])
         return container
+
+
+class LiveTests(AbstractLiveTests):
+
+    title = "Live tests (regression + validation)"
+
+    def _test_command(self) -> str:
+        selected_streams = ["--stream", self.selected_streams] if self.selected_streams else []
+        return " ".join(
+            [
+                "poetry",
+                "run",
+                "pytest",
+                "src/live_tests",
+                "--connector-image",
+                self.connector_image,
+                "--connection-id",
+                self.connection_id or "",
+                "--control-version",
+                self.control_version or "",
+                "--target-version",
+                self.target_version or "",
+                "--pr-url",
+                self.pr_url or "",
+                "--run-id",
+                self.run_id or "",
+                "--should-read-with-state",
+                str(self.should_read_with_state),
+                "--test-evaluation-mode",
+                str(self.test_evaluation_mode),
+            ]
+            + selected_streams
+        )
+
+
+class RegressionTests(AbstractLiveTests):
+    """A step to run regression tests for a connector."""
+
+    title = "Regression tests"
+
+    def _test_command(self) -> str:
```

Comment:
nit: I'd recommend returning the a list and joining it at call time. Just because the usual `with_exec` input is a list. (the fact we join it is an implementation detail at the container execution logic level)